### PR TITLE
Added default timeout override for e2e tests

### PR DESCRIPTION
**Pull request by zhongruige**
### Changes proposed in this Pull Request:

This PR introduces a new environment variable, `DEFAULT_TIMEOUT_OVERRIDE`, which can be used to override the default Jest timeout. This is beneficial for multiple e2e tests that may fail due to hitting the default 30000ms timeout, but without needing to run the tests in dev mode. 

### How to test the changes in this Pull Request:

1. Verify the CI passes
2. Make sure the changes look good, including the timeout selected.

### CODE REVIEW

**Comment by rrennick** [comment]
@zhongruige This looks like it should work but local testing still shows the default 30 second timeout. I dumped `testEnvVars` to verify `DEFAULT_TIMEOUT_OVERRIDE` was being observed:

```
$ DEFAULT_TIMEOUT_OVERRIDE=35000 npx wc-e2e test:e2e
...
{ NODE_ENV: 'test-e2e',
  NODE_CONFIG_DIR:
   '/Users/ronrennick/Sites/solaris/woocommerce/tests/e2e/config:/Users/ronrennick/Sites/solaris/woocommerce/tests/e2e/env/config',
  node_config_dev: 'no',
  jest_test_timeout: '35000',
  JEST_PUPPETEER_CONFIG:
   '/Users/ronrennick/Sites/solaris/woocommerce/tests/e2e/env/config/jest-puppeteer.config.js' }
```

```
 TimeoutError: waiting for selector ".updated.notice" failed: timeout 30000ms exceeded
```

**Comment by zhongruige** [comment]
> @zhongruige This looks like it should work but local testing still shows the default 30 second timeout. I dumped `testEnvVars` to verify `DEFAULT_TIMEOUT_OVERRIDE` was being observed:
> 
> ```
> $ DEFAULT_TIMEOUT_OVERRIDE=35000 npx wc-e2e test:e2e
> ...
> { NODE_ENV: 'test-e2e',
>   NODE_CONFIG_DIR:
>    '/Users/ronrennick/Sites/solaris/woocommerce/tests/e2e/config:/Users/ronrennick/Sites/solaris/woocommerce/tests/e2e/env/config',
>   node_config_dev: 'no',
>   jest_test_timeout: '35000',
>   JEST_PUPPETEER_CONFIG:
>    '/Users/ronrennick/Sites/solaris/woocommerce/tests/e2e/env/config/jest-puppeteer.config.js' }
> ```
> 
> ```
>  TimeoutError: waiting for selector ".updated.notice" failed: timeout 30000ms exceeded
> ```

Thanks for the comment @rrennick and review!

Interesting. I wonder if the `page.waitForSelector()` has a timeout that is overriding the Jest global one? That has a default of `30000ms` as I recall. I wonder if we need to also look at setting a global Puppeteer timeout as well, something like:

```
page.setDefaultTimeout(DEFAULT_TIMEOUT_OVERRIDE);
```

And we could set it to the same value. This also makes me wonder if we should considering making use of this timeout flag as well:

```
page.setDefaultNavigationTimeout();
```

Then also pass in the `DEFAULT_TIMEOUT_OVERRIDE` to that.

**Comment by zhongruige** [comment]
I pushed up a change introducing this logic: https://github.com/woocommerce/woocommerce/pull/30492/files#diff-85539f86ab3033e56932d26a10444f68a83ac73278b00bd545e8cbd147dea92dR43

In testing, I verified Puppeteer is now using the override value:

```
    TimeoutError: waiting for selector ".updated.notice" failed: timeout 120000ms exceeded
```

File: tests/e2e/env/bin/e2e-test-integration.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const program = require( 'commander' );
 const path = require( 'path' );
 const fs = require( 'fs' );
 const { getAppRoot } = require( '../utils' );
-const { WC_E2E_SCREENSHOTS, JEST_PUPPETEER_CONFIG } = process.env;
+const { WC_E2E_SCREENSHOTS, JEST_PUPPETEER_CONFIG, DEFAULT_TIMEOUT_OVERRIDE } = process.env;
```
Comment (rrennick) [inline tests/e2e/env/bin/e2e-test-integration.js:8]: We should have this documented in the readme similar to https://github.com/woocommerce/woocommerce/blob/19c193870016670e9486ddd795b9cdcecec703f4/tests/e2e/env/README.md#test-screenshots and have a changelog entry

**Review by rrennick — CHANGES_REQUESTED** [review]
@zhongruige Nice work on this. I saw the correct timeout value after your latest changes:

```
https://github.com/woocommerce/woocommerce/blob/19c193870016670e9486ddd795b9cdcecec703f4/tests/e2e/env/README.md#test-screenshots
```

This should be good to go once the documentation updates have been added.

File: tests/e2e/env/bin/e2e-test-integration.js (hunk 1 above)
Comment (zhongruige) [reply tests/e2e/env/bin/e2e-test-integration.js:8]: Good call out. I've added those in here:

* Changelog: https://github.com/woocommerce/woocommerce/pull/30492/files#diff-713add7c2ab3532b3a59dc88e5745b85ab74ac1e94c9a144afc0a6ac656e1bf0R10
* README: https://github.com/woocommerce/woocommerce/pull/30492/files#diff-fb9ea0584cd5ee98bcca789f51520e4c7950d7d6a1785d81d2d81e39d71a94aeR90

**Review by rrennick — APPROVED** [review]
@zhongruige This is a nice fix & enhancement.

**Pull request merged**